newIDE/app/src/AssetStore/index.js

```diff
@@ -600,72 +600,78 @@ export const AssetStore = React.forwardRef<Props, AssetStoreInterface>(
 
     return (
       <Column expand noMargin useFullHeight noOverflowParent id="asset-store">
-        <LineStackLayout>
-          <IconButton
-            id="home-button"
-            key="back-discover"
-            tooltip={t`Back to discover`}
-            onClick={() => {
-              setSearchText('');
-              const page = assetSwappedObject
-                ? shopNavigationState.openAssetSwapping()
-                : shopNavigationState.openHome();
-              setScrollUpdateIsNeeded(page);
-              clearAllAssetStoreFilters();
-              setIsFiltersPanelOpen(false);
-            }}
-            size="small"
-          >
-            <Home />
-          </IconButton>
-          <Column expand useFullHeight noMargin>
-            <SearchBar
-              placeholder={
-                hideGameTemplates ? t`Search assets` : `Search the shop`
-              }
-              value={searchText}
-              onChange={(newValue: string) => {
-                if (searchText === newValue) {
-                  return;
+        <>
+          <LineStackLayout>
+            {!assetSwappedObject && (
+              <IconButton
+                id="home-button"
+                key="back-discover"
+                tooltip={t`Back to discover`}
+                onClick={() => {
+                  setSearchText('');
+                  const page = assetSwappedObject
+                    ? shopNavigationState.openAssetSwapping()
+                    : shopNavigationState.openHome();
+                  setScrollUpdateIsNeeded(page);
+                  clearAllAssetStoreFilters();
+                  setIsFiltersPanelOpen(false);
+                }}
+                size="small"
+              >
+                <Home />
+              </IconButton>
+            )}
+            <Column expand useFullHeight noMargin>
+              <SearchBar
+                placeholder={
+                  hideGameTemplates ? t`Search assets` : `Search the shop`
                 }
-                setSearchText(newValue);
-                if (isOnSearchResultPage) {
-                  // An existing search is already being done: just move to the
-                  // top search results.
-                  shopNavigationState.openSearchResultPage();
-                  const assetsListInterface = assetsList.current;
-                  if (assetsListInterface) {
-                    assetsListInterface.scrollToPosition(0);
-                    assetsListInterface.setPageBreakIndex(0);
+                value={searchText}
```

Comment:
we should probably initiate the searchText to something similar to the object being swapped, to get better results.
Though, some names may not give results at all

Reply:
I think there should be ordering based on the tag already

Reply:
ok, leaving as is then.